readthedocs/core/views.py

```diff
@@ -811,3 +812,75 @@ def get_results(self):
         Fetches the results via the form.
         """
         return self.form.search()
+
+
+class SendEmailView(FormView):
+
+    """Form view for sending emails to users from admin pages
+
+    Accepts the following additional parameters:
+
+    queryset
+        The queryset to use to determine the users to send emails to
+    """
+
+    form_class = SendEmailForm
+    template_name = 'core/send_email_form.html'
+
+    def get_form_kwargs(self):
+        """Override form kwargs based on input fields
+
+        The admin posts to this view initially, so detect the send button on
+        form post variables. Drop additional fields if we see the send button.
+        """
+        kwargs = super(SendEmailView, self).get_form_kwargs()
+        if 'send' not in self.request.POST:
+            kwargs.pop('data', None)
+            kwargs.pop('files', None)
+        return kwargs
+
+    def get_initial(self):
+        """Add selected ids to initial form data"""
+        initial = super(SendEmailView, self).get_initial()
+        initial['_selected_action'] = self.request.POST.getlist(
+            admin.ACTION_CHECKBOX_NAME)
+        return initial
+
+    def form_valid(self, form):
+        """If form is valid, send emails to selected users"""
+        count = 0
+        for user in self.get_queryset().all():
+            send_email(
+                user.email,
+                subject=form.cleaned_data['subject'],
+                template='core/email/common.txt',
+                template_html='core/email/common.html',
+                context={'user': user, 'content': form.cleaned_data['body']},
+                request=self.request,
+            )
+            count += 1
+        if count == 0:
+            self.message_user("No receipients to send to", level=messages.ERROR)
+        else:
+            self.message_user("Messages sent!")
+        return HttpResponseRedirect(self.request.get_full_path())
+
+    def get_queryset(self):
+        return self.kwargs.get('queryset')
+
+    def get_context_data(self, **kwargs):
+        """Return queryset in context"""
+        context = super(SendEmailView, self).get_context_data(**kwargs)
+        context['users'] = self.get_queryset().all()
+        return context
+
+    def message_user(self, message, level=messages.INFO, extra_tags='',
+                     fail_silently=False):
+        """Implementation of :py:meth:`django.contrib.admin.options.ModelAdmin.message_user`
+
+        Send message through messages framework
+        """
+        # TODO generalize this or check if implementation in ModelAdmin is
+        # useable here
+        messages.add_message(self.request, level, message, extra_tags=extra_tags,
+                             fail_silently=fail_silently)
```

Comment:
This method was included in the base code, it should be removed in the future and maybe just passed back to the `ModelAdmin` instance, if we pass that in to the view instantiation.